commit 34199a27257d2f764946b625aeaf785f14f59ae6
Author: John Levon <john.levon@joyent.com>
Date:   2019-07-08T13:43:07+00:00 (3 months ago)
    
    MANTA-4383 update muppet to triton-origin-x86_64-18.4.0
    MANTA-4394 clean up some muppet droppings

diff --git a/.gitignore b/.gitignore
index 81afecd..e81f14e 100644
--- a/.gitignore
+++ b/.gitignore
@@ -9,5 +9,6 @@ etc/haproxy.cfg
 cscope.in.out
 cscope.po.out
 cscope.out
+package-lock.json
 smf/manifests/*.xml
 /*.tar.gz
diff --git a/Makefile b/Makefile
index e38920e..5031a85 100644
--- a/Makefile
+++ b/Makefile
@@ -5,7 +5,7 @@
 #
 
 #
-# Copyright (c) 2019, Joyent, Inc.
+# Copyright 2019 Joyent, Inc.
 #
 
 #
@@ -45,12 +45,12 @@ SMF_MANIFESTS_IN = smf/manifests/$(NAME).xml.in smf/manifests/haproxy.xml.in smf
 # Variables
 #
 
-NODE_PREBUILT_TAG	= zone
-NODE_PREBUILT_VERSION   := v4.6.1
-NODE_PREBUILT_IMAGE     = 18b094b0-eb01-11e5-80c1-175dac7ddf02
+NODE_PREBUILT_VERSION=v6.17.0
+NODE_PREBUILT_TAG=zone64
+NODE_PREBUILT_IMAGE=c2c31b00-1d60-11e9-9a77-ff9f06554b0f
 
-ENGBLD_USE_BUILDIMAGE	= true
-ENGBLD_REQUIRE		:= $(shell git submodule update --init deps/eng)
+ENGBLD_USE_BUILDIMAGE = true
+ENGBLD_REQUIRE := $(shell git submodule update --init deps/eng)
 include ./deps/eng/tools/mk/Makefile.defs
 TOP ?= $(error Unable to access eng.git submodule Makefiles.)
 
@@ -72,10 +72,11 @@ RELEASE_TARBALL		:= muppet-pkg-$(STAMP).tar.gz
 ROOT			:= $(shell pwd)
 RELSTAGEDIR			:= /tmp/$(NAME)-$(STAMP)
 
-BASE_IMAGE_UUID = 04a48d7d-6bb5-4e83-8c3b-e60a99e0f48f
+# our base image is triton-origin-x86_64-18.4.0
+BASE_IMAGE_UUID = a9368831-958e-432d-a031-f8ce6768d190
 BUILDIMAGE_NAME = manta-loadbalancer
 BUILDIMAGE_DESC	= Manta loadbalancer
-BUILDIMAGE_PKGSRC = py27-curses openssl-1.0.2o stud-0.3p53nb5 libzookeeper-3.4.6
+BUILDIMAGE_PKGSRC = openssl-1.0.2p stud-0.3p53nb7
 AGENTS		= amon config registrar
 
 #
@@ -92,13 +93,9 @@ CLEAN_FILES += $(NODEUNIT) ./node_modules/nodeunit
 DISTCLEAN_FILES += ./node_modules
 
 .PHONY: test
-#
-# Unit tests
-# Right now we are ignoring the watch test until
-# it is back to a working state.
-#
+
 test: $(NODEUNIT)
-	$(NODEUNIT) test/config.test.js 2>&1 | $(BUNYAN)
+	$(NODEUNIT) test/*.test.js 2>&1 | $(BUNYAN)
 
 .PHONY: scripts
 scripts: deps/manta-scripts/.git
diff --git a/deps/eng b/deps/eng
index ed10150..221cb14 160000
--- a/deps/eng
+++ b/deps/eng
@@ -1 +1 @@
-Subproject commit ed10150657b1f3ae4f061c751daf365ed16ca8d5
+Subproject commit 221cb144d090e0ab81b915bdea38e5a43b4a5a7f
diff --git a/env.sh b/env.sh
deleted file mode 100644
index 9017cb0..0000000
--- a/env.sh
+++ /dev/null
@@ -1,13 +0,0 @@
-#
-# This Source Code Form is subject to the terms of the Mozilla Public
-# License, v. 2.0. If a copy of the MPL was not distributed with this
-# file, You can obtain one at http://mozilla.org/MPL/2.0/.
-#
-
-#
-# Copyright (c) 2017, Joyent, Inc.
-#
-
-export PATH=$PWD/build/node/bin:$PWD/node_modules/.bin:$PATH
-
-alias muppet='sudo node muppet.js -f ./etc/config.coal.json -v 2>&1 | bunyan'
diff --git a/etc/config.coal.json b/etc/config.coal.json
deleted file mode 100644
index c54759e..0000000
--- a/etc/config.coal.json
+++ /dev/null
@@ -1,13 +0,0 @@
-{
-        "name": "manta.coal.joyent.us",
-        "trustedIP": "127.0.0.1",
-        "untrustedIPs": [ "::1" ],
-        "zookeeper": {
-                "servers": [ {
-                        "host": "10.99.99.11",
-                        "port": 2181
-                } ],
-                "timeout": 60000
-        },
-        "restart": "/usr/bin/true"
-}
diff --git a/package.json b/package.json
index f9522a5..e164460 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "muppet",
   "description": "Joyent's Load Balancer",
-  "version": "1.1.0",
+  "version": "1.2.0",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
diff --git a/test/watch.test.js b/test/watch.test.js
deleted file mode 100644
index e5192bb..0000000
--- a/test/watch.test.js
+++ /dev/null
@@ -1,143 +0,0 @@
-/*
- * This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/.
- */
-
-/*
- * Copyright (c) 2014, Joyent, Inc.
- */
-
-var uuid = require('node-uuid');
-var vasync = require('vasync');
-var zkplus = require('zkplus');
-
-var core = require('../lib');
-
-if (require.cache[__dirname + '/helper.js'])
-        delete require.cache[__dirname + '/helper.js'];
-var helper = require('./helper.js');
-
-
-
-///--- Globals
-
-var after = helper.after;
-var before = helper.before;
-var test = helper.test;
-
-var DOMAIN = 'watch.unit.test';
-var PATH = '/test/unit/watch';
-
-var WATCH;
-var ZK;
-var ZK2; // We need 2 ZK clients as ZK has crazy ordering of watches vs commits
-
-
-///--- Tests
-
-test('create zk', function (t) {
-        helper.createZkClient(function (err, zk) {
-                t.ifError(err);
-                t.ok(zk);
-                ZK = zk;
-                t.end();
-        });
-});
-
-
-test('create zk2', function (t) {
-        helper.createZkClient(function (err, zk) {
-                t.ifError(err);
-                t.ok(zk);
-                ZK2 = zk;
-                t.end();
-        });
-});
-
-test('create watch', function (t) {
-        WATCH = core.createWatch({
-                domain: DOMAIN,
-                log: helper.createLogger(),
-                zk: ZK
-        });
-        t.ok(WATCH);
-        WATCH.start(function (err) {
-                t.ifError(err);
-                WATCH.once('hosts', function (hosts) {
-                        t.ok(hosts);
-                        t.equal(hosts.length, 0);
-                        t.end();
-                });
-        });
-});
-
-
-test('host addition', function (t) {
-        WATCH.once('hosts', function (hosts) {
-                t.ok(hosts);
-                t.equal(hosts.length, 1);
-                t.equal(hosts[0], '192.168.0.1');
-                t.end();
-        });
-
-        var opts = {
-                object: {
-                        type: 'host',
-                        host: {
-                                address: '192.168.0.1'
-                        }
-                }
-        };
-        var p = PATH + '/1';
-        ZK2.creat(p, opts, function (err) {
-                t.ifError(err);
-        });
-});
-
-
-test('host drop out', function (t) {
-        var obj = {
-                type: 'host',
-                host: {
-                        address: '192.168.0.2'
-                }
-        };
-        ZK2.put(PATH + '/2', obj, function (err) {
-                t.ifError(err);
-
-                // We need to wait here otherwise the ZK
-                // library will fire on the next invocation for the previous
-                // add (i.e., when we set this watch, it potentially gets set
-                // *before* the watch logic is done for the put above).
-                setTimeout(function () {
-                        WATCH.once('hosts', function (hosts) {
-                                t.ok(hosts);
-                                t.equal(hosts.length, 1);
-                                t.equal(hosts[0], '192.168.0.2');
-                                t.end();
-                        });
-
-                        ZK2.rmr(PATH + '/1', function (err2) {
-                                t.ifError(err2);
-                        });
-                }, 500);
-        });
-});
-
-
-test('tear down', function (t) {
-        ZK.on('close', function () {
-                ZK2.rmr('/test', function (err) {
-                        ZK2.on('close', function () {
-                                t.end();
-                        });
-                        ZK2.close();
-                });
-        });
-
-        WATCH.stop();
-        process.nextTick(function () {
-                ZK.close();
-        });
-});
